Guidance on fixing TypeScript type-definitions

* Output of `node_modules/.bin/electron --version`: v1.8.8
* Operating System (Platform and Version): macOS

**Expected Behavior**
When reading up the docs for [BrowserWindow.setBounds](https://electronjs.org/docs/api/browser-window#winsetboundsbounds-animate) I expected the function to take a `Partial<Rectangle>` as first argument

**Actual behavior**
but instead the generated type definitions are:

```typescript
/**
  * Resizes and moves the window to the supplied bounds
  */
setBounds(bounds: Rectangle, animate?: boolean): void;
```

Now what I'd be really looking for is some minor instruction on how to contribute such fixes by myself. Looking through the [package.json of electron](https://github.com/electron/electron/blob/master/package.json#L57) I noticed the use of a [custom dependency](https://github.com/electron/electron-typescript-definitions) to generate the definitions out of a previously generated api.json from this [module](electron-docs-linter). Now the later kinda sounds like the fix needs to be done inside the documentation. So after digging around [the docs for the BrowserWindow](https://raw.githubusercontent.com/electron/electron/5ae3d1a1b2dbe11d3091d366467591d9cb21fdfe/docs/api/browser-window.md) show this content to define the method:

```
#### `win.setBounds(bounds[, animate])`

* `bounds` [Rectangle](structures/rectangle.md)
* `animate` Boolean (optional) _macOS_
```

Am I on the right track with my assumptions? If so how would I mark the Rectangle of the bounds property to be Partial. Are there any docs to see all the available constructs to be used as annotation or which features of TS are supported (think of Unions, Conditional Types etc.)

@MarshallOfSound ok I've tried out your last comments approach, which fails with:
```
index.d.ts(1759,23): error TS2314: Generic type 'Partial' requires 1 type argument(s).
```

and creates this as a resulting d.ts

```typescript
/**
     * Resizes and moves the window to the supplied bounds. Any properties that are not
     * supplied will default to their current values.
     */
    setBounds(bounds: Partial, animate?: boolean): void;
```

Note the lack of the generic type argument.

--- a/docs/api/browser-window.md
+++ b/docs/api/browser-window.md
@@ -978,7 +978,7 @@ Closes the currently open [Quick Look][quick-look] panel.
 
 #### `win.setBounds(bounds[, animate])`
 
-* `bounds` [Rectangle](structures/rectangle.md)
+* `bounds` Partial<[Rectangle](structures/rectangle.md)>
 * `animate` Boolean (optional) _macOS_
 
 Resizes and moves the window to the supplied bounds. Any properties that are not supplied will default to their current values.

